### PR TITLE
Fix #948: feat(auto-merge): enable safe auto-merge for paid-agent PR reviews

### LINKED ISSUE #948: feat(auto-merge): enable safe auto-merge for paid-agent PR reviews
## Summary

Enable fully automatic PR review and merge using Paid agent reviews (paid_agent review method), achieving parity with the Copilot and Codex review workflows that already work reliably with auto-merge.

Currently auto-merge is manually disabled (#824) because the scanner, review triggering, and merge gates do not account for paid-agent reviews. This umbrella issue tracks all prerequisite work to safely re-enable auto-merge with paid_agent reviews.

## Strategy

The approach centers on creating a dedicated GitHub bot account for paid-agent reviews. Registering the bot in PROVIDER_BOT_USERNAMES collapses three scanner gaps (identity confusion, merge gate, escalation) because the existing bot-detection infrastructure handles the new login automatically.

### Infrastructure

- viamin/paid#947 — Create the paid-agent reviewer bot account and wire token into Paid
- viamin/paid#945 — Route review-goal proxy requests through the bot identity

### Scanner trigger and detection

- viamin/paid#915 — Auto-trigger paid_agent reviews in the auto-continue loop (trigger routing still needed; detection side solved by bot account)
- viamin/paid#918 — Clean signal detection for paid_agent reviews (define a machine-readable marker)
- viamin/paid#914 — Draft phase exit gate for paid_agent (simplified by bot account; needs #915 and #918)
- viamin/paid#944 — Enforce paid_agent max_review_rounds to prevent infinite review loops

### Reliability

- viamin/paid#830 — Fix review-goal idle timeouts that stall draft PR progression
- viamin/paid#824 — Comprehensive auto-merge gate (block merge until all review signals complete)

### Scanner coverage

- viamin/paid#919 — Draft exit and ready detection ignore manual and ci_action review methods

### Auto-merge safety

- viamin/paid#934 — Require clean re-review on HEAD commit for body-only bot clearance
- viamin/paid#935 — Check affected files when clearing body-only review feedback

### Quality metrics

- viamin/paid#943 — Webhook feedback attribution misses review-goal agent runs

### Audit and monitoring

- viamin/paid#825 — Remediate paid PRs merged before review completion
- viamin/paid#913 — Operational alert rules for stalled PRs, runaway loops, quota exhaustion

## Acceptance criteria

- [ ] paid_agent review method can be enabled on a project
- [ ] Review-goal agent runs are automatically triggered when paid_agent is enabled
- [ ] Reviews appear under the paid-agent bot identity on GitHub
- [ ] Scanner correctly classifies bot reviews (clean vs. has-findings)
- [ ] Draft phase does not advance until a clean review or max rounds reached
- [ ] Auto-merge only fires after all review gates pass (no unresolved threads, clean review signal, CI green)
- [ ] Review rounds are capped at max_review_rounds with escalation to owner
- [ ] Existing Copilot and Codex review workflows are unaffected
- [ ] Auto-merge is re-enabled on paid after all P1 issues are resolved and P1 audit (#825) is complete

## Out of scope

- Manual and ci_action review method gating (#919 is P2)
- Notification/alerting rules (#913 is P2)
- Quality metrics for review-goal runs (#943 is P2)

## Related

- viamin/paid#942 — Ensure planning label exists in GitHub repo for auto-pick exclusion
- viamin/paid#826 — Reconcile internal pr_review_phase with manual GitHub draft/ready transitions

### PR DESCRIPTION
**Pull request by viamin**
## Summary

Enable the paid-agent review method to participate in the auto-merge pipeline by registering a dedicated bot identity, wiring up clean-signal detection, and auto-triggering review-goal runs — closing the three scanner gaps (identity confusion, merge gate, escalation) that blocked auto-merge for paid-agent reviews.

This is the core P1 implementation for #838, bringing paid-agent reviews to parity with Copilot and Codex review workflows that already work reliably with auto-merge.

### Infrastructure — Bot identity registration
- Registered `paid-agent` and `paid-agent[bot]` logins in `PROVIDER_BOT_USERNAMES`, so the existing bot-detection infrastructure (`review_bot?()`, `enabled_review_bot_logins`, `bot_user?()`) automatically recognizes paid-agent reviews without any scanner-specific logic.

### Scanner — Clean signal detection
- Extended `REVIEW_BOT_CLEAN_PATTERN` to match the `<!-- paid-review-signal: clean -->` HTML comment marker, giving the scanner a machine-readable way to distinguish clean reviews from reviews with findings.

### Trigger routing — Auto-trigger paid-agent reviews
- Added `PAID_AGENT_LOGIN` constant and wired `review_bot_request_login` to return it when `paid_agent` is the configured review method.
- When `review_bot_review_pending` fires with the paid-agent login, the poll workflow routes to a new `start_review_goal_workflow` path that queues a review-goal agent run instead of calling `RequestReviewActivity` (which handles external bot APIs).
- `QueueAgentRunActivity` now accepts an optional `goal` parameter for programmatic review-goal runs.

### Agent prompt — Clean signal instructions
- Injected instructions into the review agent prompt to emit `<!-- paid-review-signal: clean -->` in the review body when no issues are found, closing the detection loop end-to-end.

## Design Decisions

- **Bot account in `PROVIDER_BOT_USERNAMES` as the keystone change**: Rather than adding paid-agent-specific conditionals throughout the scanner, registering the bot login collapses identity detection, merge-gate eligibility, and escalation handling into the existing bot infrastructure. This is a deliberate choice to minimize surface area.
- **HTML comment as clean signal**: Using `<!-- paid-review-signal: clean -->` keeps the marker invisible to humans reading the review on GitHub while being trivially parseable by the scanner regex. This mirrors the pattern already used by other review bot integrations.
- **Routing divergence for paid-agent vs. external bots**: External bots (Copilot, Codex) use `RequestReviewActivity` to call provider APIs, but paid-agent reviews are internal — they queue an agent run directly. The `start_review_goal_workflow` branch keeps this distinction explicit rather than overloading the request-review path.

---

Closes #948

### CODE REVIEW

**Review by viamin — COMMENTED** [review]
## Review Summary

Clean, well-scoped PR that wires paid-agent reviews into the existing auto-merge infrastructure. The approach of registering in `PROVIDER_BOT_USERNAMES` to reuse existing bot detection is the right call — minimal surface area.

Two substantive concerns below (one on deduplication, one on the regex), plus a minor style nit. Otherwise LGTM.

File: app/temporal/activities/queue_agent_run_activity.rb [hunk 1]
```diff
@@ -31,15 +32,17 @@ def execute(input)
...
             project: project,
             issue: issue,
             provider_id: provider_id,
             agent_type: agent_type,
             custom_prompt: custom_prompt,
             source_pull_request_number: source_pull_request_number,
             status: "queued"
-          )
+          }
+          attrs[:goal] = goal if goal.present?
```
Comment (viamin) [inline app/temporal/activities/queue_agent_run_activity.rb]: **Potential dedup conflict between review and non-review runs.**

`find_existing_run` matches by issue/PR number without considering `goal`. If a `create_pr` run is already in-flight for the same issue (e.g., a followup agent fixing code), then a subsequent `start_review_goal_workflow` call will hit the dedup path and return `{ duplicate: true }` — the review-goal run never gets queued.

This is the same race the PR description calls out ("defer the review request … so the bot reviews the fixed code"), but the `handle_review_bot_review_pending` guard only covers the case where *other triggers* exist in the same poll cycle. If a followup run was queued in a *previous* cycle and is still running, the review will be silently dropped.

Consider scoping `find_existing_run` to also match on `goal`, or at minimum documenting this as a known limitation:

```suggestion
          attrs[:goal] = goal if goal.present?
          # NOTE: find_existing_run does not scope by goal, so a review-goal
          # run will be deduplicated against an in-flight create_pr run for
          # the same issue/PR. This is acceptable for now because the poll
          # workflow defers review requests when other triggers are present,
          # but could miss reviews if a prior-cycle followup is still running.
          run = AgentRun.create!(**attrs)
```

File: app/temporal/activities/scan_paid_prs_activity.rb [hunk 2]
```diff
@@ -20,7 +20,7 @@ class ScanPaidPrsActivity < BaseActivity
 
     MIN_COMMENT_LENGTH = 20
     KNOWN_BOT_PREFIXES = %w[dependabot renovate github-actions].freeze
-    REVIEW_BOT_CLEAN_PATTERN = /generated no (?:new )?comments/i
+    REVIEW_BOT_CLEAN_PATTERN = /generated no (?:new )?comments|<!-- paid-review-signal: clean -->/i
```
Comment (viamin) [inline app/temporal/activities/scan_paid_prs_activity.rb]: **Regex alternation precedence — the HTML comment branch will never match with `/i`.**

This isn't actually a bug (the regex works correctly since `|` has the lowest precedence in regex), but the case-insensitive flag on the HTML comment branch is worth noting: `<!-- paid-review-signal: clean -->` is a machine-emitted literal, so matching it case-insensitively could allow false positives if user-authored review text happens to contain a similar string. Consider anchoring the marker or using a case-sensitive match for the new branch.

Minor concern — not blocking.

File: app/temporal/workflows/git_hub_poll_workflow.rb [hunk 3]
```diff
@@ -328,5 +333,12 @@ def start_pr_followup_workflow(project_id, pr_data)
           source_pull_request_number: pr_number }, timeout: 30)
       run_activity(Activities::RecordPrFollowupActivity, followup_input, timeout: 30)
     end
+
+    def start_review_goal_workflow(project_id, pr_data)
+      run_activity(Activities::QueueAgentRunActivity,
+        { project_id: project_id, issue_id: pr_data[:issue_id],
+          source_pull_request_number: pr_data[:pr_number],
+          goal: "review" }, timeout: 30)
```
Comment (viamin) [inline app/temporal/workflows/git_hub_poll_workflow.rb:398]: **Missing `RecordDraftReviewActivity` for review-goal runs.**

The existing `start_draft_followup_workflow` calls `RecordDraftReviewActivity` after queueing the agent run, which increments the draft review round counter used to enforce `max_draft_review_rounds`. `start_review_goal_workflow` skips this, so paid-agent review-goal runs in draft phase won't count toward the max rounds cap.

If this is intentional (review-goal runs shouldn't count as draft review rounds), a comment explaining why would help. If it's not, the review loop could run indefinitely for paid-agent reviews — which is exactly what #944 (listed as a P1 dependency) is meant to prevent.

File: app/temporal/activities/run_agent_activity.rb [hunk 4]
```diff
@@ -1043,6 +1043,11 @@ def augment_prompt_for_review_goal(agent_run, prompt)
         endpoint. This is how your review is tracked as complete. Standalone PR comments
         via `/issues/#{pr_number}/comments` are optional and do NOT satisfy the review requirement.
 
+        CLEAN SIGNAL: If you find NO issues requiring action, include the following marker
+        in your review body so the automation system can detect a clean review:
+        <!-- paid-review-signal: clean -->
+        Example clean review body: "Looks good! No issues found.\\n<!-- paid-review-signal: clean -->"
```
Comment (viamin) [inline app/temporal/activities/run_agent_activity.rb]: Nit: the double-escaped `\\n` in the example will render as a literal `\n` in the agent's prompt rather than a newline. Since the agent is constructing JSON for the API call, this is probably fine (the agent will use `\n` in JSON), but it could confuse the LLM into emitting a literal backslash-n in its review body.

Consider making the example clearer:
```suggestion
        Example clean review body: "Looks good! No issues found.\n<!-- paid-review-signal: clean -->"
```

**Comment by viamin** [comment]
## Agent Update

Commit succeeded. All four review threads have been addressed:

1. **QueueAgentRunActivity dedup conflict** — Added a comment documenting the known limitation that `find_existing_run` doesn't scope by `goal`, so review-goal runs may be deduplicated against in-flight create_pr runs.

2. **Regex case-sensitivity** — Split the combined `REVIEW_BOT_CLEAN_PATTERN` into two patterns: the original case-insensitive one for existing bots, and a new case-sensitive `PAID_REVIEW_CLEAN_PATTERN` for the machine-emitted `<!-- paid-review-signal: clean -->` marker.

3. **Missing RecordDraftReviewActivity** — Added a comment explaining why `start_review_goal_workflow` intentionally skips it: review-goal runs are bot reviews (read + comment), not code-change iterations, so they shouldn't count toward `max_draft_review_rounds`.

4. **Double-escaped `\\n` in prompt** — Fixed to a single `\n` so the agent sees a real newline in the example.

File: app/temporal/activities/request_review_activity.rb [hunk 5]
```diff
@@ -25,6 +25,7 @@ class RequestReviewActivity < BaseActivity
 
     COPILOT_LOGIN = "copilot"
     CODEX_LOGIN = "chatgpt-codex-connector"
```
Comment (viamin) [inline app/temporal/activities/request_review_activity.rb:27]: `PAID_AGENT_LOGIN` is defined here in `RequestReviewActivity`, but paid-agent reviews are explicitly routed *away* from this activity — they go through `start_review_goal_workflow` → `QueueAgentRunActivity` instead. This makes the constant's home misleading.

Consider moving it to a shared location (e.g., `ProviderSupport` alongside `PROVIDER_BOT_USERNAMES`, or a dedicated constants module) so the reference in `Project#review_bot_request_login` and `GitHubPollWorkflow` doesn't imply a dependency on `RequestReviewActivity`.

Not blocking, but worth addressing to avoid confusion as more review methods are added.

File: app/temporal/activities/queue_agent_run_activity.rb [hunk 6]
```diff
@@ -31,15 +32,22 @@ def execute(input)
...
```
Comment (viamin) [inline app/temporal/activities/queue_agent_run_activity.rb]: The conditional `attrs[:goal] = goal if goal.present?` means that when `goal` is nil (the common `create_pr` path), the column's DB default (`"create_pr"`) kicks in. This works, but it's fragile — it couples the Ruby code to the DB default rather than being explicit.

Consider:
```suggestion
          attrs[:goal] = goal.presence || "create_pr"
```
This makes the default visible in the code and avoids surprises if the DB default ever changes.

File: app/temporal/activities/queue_agent_run_activity.rb [hunk 7]
```diff
@@ -31,15 +32,22 @@ def execute(input)
...
+          # NOTE: find_existing_run does not scope by goal, so a review-goal
+          # run will be deduplicated against an in-flight create_pr run for
+          # the same issue/PR. This is acceptable for now because the poll
+          # workflow defers review requests when other triggers are present,
+          # but could miss reviews if a prior-cycle followup is still running.
```
Comment (viamin) [inline app/temporal/activities/queue_agent_run_activity.rb:56]: This comment accurately flags a real gap: if a `create_pr` followup is still running when the poll workflow fires `review_bot_review_pending`, `find_existing_run` will match that in-flight run and silently skip the review-goal enqueue.

The poll workflow's deferral logic (checking `other_triggers`) helps, but there's a race window: a prior-cycle followup could still be `queued`/`active` when a new poll cycle finds no other triggers and tries to enqueue the review.

Worth tracking as a follow-up issue (or linking to an existing one) so this doesn't get lost. The comment alone may not surface during future debugging.

**Review by viamin — COMMENTED** [review]
## Review Summary

Clean, well-structured PR that integrates paid-agent reviews into the auto-merge pipeline by reusing existing bot infrastructure. The approach of registering the bot identity in `PROVIDER_BOT_USERNAMES` to collapse three scanner gaps is sound and minimizes surface area.

**Key strengths:**
- Good use of existing bot-detection infrastructure instead of adding paid-agent-specific conditionals
- Helpful inline comments explaining deduplication trade-offs and the decision to skip `RecordDraftReviewActivity`
- Clean separation between internal (paid-agent) and external (Copilot/Codex) review routing

**Issues found:**
1. **No tests** — This is the most significant gap. The new routing path, clean signal detection, and goal parameter are all untested.
2. **Constant placement** — `PAID_AGENT_LOGIN` lives in `RequestReviewActivity` but paid-agent reviews never flow through that activity.
3. **Deduplication gap** — The noted `find_existing_run` limitation could silently drop reviews.

See inline comments for details.

File: app/temporal/activities/scan_paid_prs_activity.rb [hunk 8]
```diff
@@ -21,6 +21,10 @@ class ScanPaidPrsActivity < BaseActivity
     MIN_COMMENT_LENGTH = 20
     KNOWN_BOT_PREFIXES = %w[dependabot renovate github-actions].freeze
     REVIEW_BOT_CLEAN_PATTERN = /generated no (?:new )?comments/i
+    # Separate case-sensitive pattern for the machine-emitted paid-agent
+    # clean signal. Case-sensitivity avoids false positives from
+    # user-authored review text that might resemble the marker.
+    PAID_REVIEW_CLEAN_PATTERN = /<!-- paid-review-signal: clean -->/
```
Comment (viamin) [inline app/temporal/activities/scan_paid_prs_activity.rb]: Having two separate patterns (`REVIEW_BOT_CLEAN_PATTERN` and `PAID_REVIEW_CLEAN_PATTERN`) with different case sensitivity is a reasonable choice, but the naming divergence (`REVIEW_BOT_CLEAN_PATTERN` vs `PAID_REVIEW_CLEAN_PATTERN`) breaks the naming convention — the paid-agent *is* a review bot.

Consider a name like `REVIEW_BOT_PAID_CLEAN_PATTERN` or documenting on `REVIEW_BOT_CLEAN_PATTERN` that it covers external bots while the paid pattern covers internal ones. Minor, but helps future readers understand the taxonomy.

File: app/temporal/workflows/git_hub_poll_workflow.rb [hunk 9]
```diff
@@ -328,5 +333,17 @@ def start_pr_followup_workflow(project_id, pr_data)
...
+    # they are bot reviews (reading + commenting), not code-change iterations.
+    # Draft review rounds track code-fix cycles; counting a review-goal run
+    # would inflate the counter and prematurely trigger escalation. The
+    # max_draft_review_rounds cap (#944) applies to create_pr followups only.
+    def start_review_goal_workflow(project_id, pr_data)
+      run_activity(Activities::QueueAgentRunActivity,
+        { project_id: project_id, issue_id: pr_data[:issue_id],
+          source_pull_request_number: pr_data[:pr_number],
+          goal: "review" }, timeout: 30)
+    end
```
Comment (viamin) [inline app/temporal/workflows/git_hub_poll_workflow.rb:399]: **Missing tests**: This new `start_review_goal_workflow` method and the routing branch in `handle_review_bot_review_pending` are the core behavioral changes of this PR, but no specs were added.

At minimum, I'd expect:
1. A workflow spec verifying that `handle_review_bot_review_pending` routes to `start_review_goal_workflow` when login is `PAID_AGENT_LOGIN`
2. A workflow spec verifying it still routes to `request_review` for other logins
3. A `QueueAgentRunActivity` spec confirming the `goal` parameter is passed through to `AgentRun.create!`
4. A `ScanPaidPrsActivity` spec verifying `PAID_REVIEW_CLEAN_PATTERN` matches the expected marker

Without these, regressions in the routing logic won't be caught.

File: app/temporal/activities/run_agent_activity.rb [hunk 10]
```diff
@@ -1043,6 +1043,11 @@ def augment_prompt_for_review_goal(agent_run, prompt)
         endpoint. This is how your review is tracked as complete. Standalone PR comments
         via `/issues/#{pr_number}/comments` are optional and do NOT satisfy the review requirement.
 
+        CLEAN SIGNAL: If you find NO issues requiring action, include the following marker
+        in your review body so the automation system can detect a clean review:
+        <!-- paid-review-signal: clean -->
+        Example clean review body: "Looks good! No issues found.\n<!-- paid-review-signal: clean -->"
```
Comment (viamin) [inline app/temporal/activities/run_agent_activity.rb]: Nit: The escaped newline `\n` in the example string is a literal backslash-n in the heredoc, not an actual newline. The review agent might emit a literal `\n` instead of a line break.

Consider using an actual newline in the example:
```suggestion
        Example clean review body: "Looks good! No issues found.
<!-- paid-review-signal: clean -->"
```

**Comment by viamin** [comment]
## Agent Update

The background setup agent finished — dependencies installed fine, and the database was already prepared (tests passed successfully during the commit). All work is complete.

**Comment by viamin** [comment]
## Agent Update

Done. Here's a summary of what was accomplished:

**Merge conflict resolved** in `app/temporal/activities/run_agent_activity.rb`:
- The conflict was between the branch's inline heredoc prompt in `augment_prompt_for_review_goal` and main's refactored `Prompts::Render.call` approach (introduced by the prompts-to-table migration PR #981).
- Resolved by taking main's `Prompts::Render` version, which uses `REVIEW_GOAL_PROMPT_SLUG` with `FALLBACK_REVIEW_GOAL_PROMPT` as a template with `{{variable}}` interpolation.
- Fixed a trailing empty line lint violation (`Layout/EmptyLinesAroundMethodBody`) in the resolved method.
- The branch's `REVIEW_BOT_PAID_CLEAN_PATTERN` (`<!-- paid-review-signal: clean -->`) remains in the scanner as an additional clean signal detection path alongside main's `REVIEW_BOT_CLEAN_PATTERN` ("Generated no new comments.").

**All 655 relevant tests pass, 0 failures. Lint clean.**

**Review by viamin — COMMENTED** [review]
Generated no new comments. The PR cleanly registers the paid-agent bot identity, wires up clean-signal detection with a case-sensitive HTML comment pattern, routes review-goal runs through QueueAgentRunActivity instead of RequestReviewActivity, and documents the dedup limitation explicitly. Test coverage is solid for the new paths. Code follows existing patterns throughout.

**Comment by viamin** [comment]
## Agent Update

Updated the paid-agent review prompt path so clean reviews now emit both the existing `"Generated no new comments."` phrase and the new `<!-- paid-review-signal: clean -->` marker in [run_agent_activity.rb](/workspace/app/temporal/activities/run_agent_activity.rb), kept the seeded prompt template in sync in [prompts.rb](/workspace/db/seeds/prompts.rb), and expanded the coupling coverage in [seeds_prompts_spec.rb](/workspace/spec/db/seeds_prompts_spec.rb).

Validation on this workspace is mixed. `bundle exec rubocop` passed. After restoring the test DB, the focused specs for the touched areas passed functionally (`186 examples, 0 failures`) when run with `COVERAGE=false`, but the repo’s pre-commit hook still failed before commit because its full `bundle exec rspec` run hits existing test-environment issues: `db:test:purge` intermittently fails with `PG::ObjectInUse` on `agent_test`, and the full suite then reports many unrelated failures. Because that hook did not pass, no commit was created. The three changes are still staged in the worktree.

**Comment by viamin** [comment]
## Agent Update

Done. Here's a summary of what was accomplished:

**Merge conflict resolved** in `spec/temporal/workflows/git_hub_poll_workflow_spec.rb`:
- Kept both the branch's paid-agent routing tests (review_bot_review_pending → paid-agent, review_bot_review_pending → copilot) and main's new tests (manual_review_pending, ci_action_pending, followup with combined triggers).

**Bug fix** in `app/temporal/activities/scan_paid_prs_activity.rb`:
- The `paid_agent` clean signal bypass in `check_review_bot_status` was broken because paid_agent is now registered in `PROVIDER_BOT_USERNAMES`, making `allowed` non-empty. The bypass condition now subtracts paid_agent logins from `allowed` before checking emptiness, so the clean marker correctly short-circuits when no external bots (copilot, codex) are configured.

**Test updates** in `spec/temporal/activities/scan_paid_prs_activity_spec.rb`:
- Two specs expecting `ready_for_owner` were updated to expect `review_bot_review_pending` — the correct behavior now that paid_agent has bot logins and the scanner properly requests a review when no clean marker is present.

**Validation**: 5758 examples, 0 failures, lint clean.

**Review by viamin — COMMENTED** [review]
I found two correctness issues in the new paid-agent review flow. Both affect mixed review-method configurations, so the PR does not yet fully satisfy the goal of making paid-agent participate safely alongside the existing bot workflows.

File: app/models/project.rb [hunk 11]
```diff
@@ -478,6 +478,7 @@ def review_bot_request_login
     return nil unless review_enabled?
     return Activities::RequestReviewActivity::COPILOT_LOGIN if review_method_enabled?("copilot")
     return Activities::RequestReviewActivity::CODEX_LOGIN if review_method_enabled?("codex")
+    return ProviderSupport::PAID_AGENT_LOGIN if review_method_enabled?("paid_agent")
```
Comment (viamin) [inline app/models/project.rb]: Returning a single `request_login` here means `paid_agent` is never requested when a higher-priority requestable bot is also enabled. For example, with `copilot + paid_agent`, the scanner will always ask for Copilot, `handle_review_bot_review_pending` will never hit the `paid-agent` branch, and no paid-agent review-goal run is ever queued. That leaves mixed configurations without paid-agent participation even though the merge gate now treats paid-agent as an enabled review bot. Please emit/request all required review bots instead of collapsing to one login.

File: app/temporal/activities/scan_paid_prs_activity.rb [hunk 12]
```diff
@@ -576,24 +581,23 @@ def check_review_bot_status(reviews, unresolved_threads, project: nil, last_run:
...
-      # :has_comments branches to fire. The explicit check keeps that
-      # future transition safe and documents the intended semantics.
+      # paid_agent has entries in PROVIDER_BOT_USERNAMES for identity
+      # detection, but those logins don't post reviews the way copilot/
+      # codex do, so we subtract them before checking whether any other
+      # bot logins remain.
+      paid_agent_logins = ProviderSupport::PROVIDER_BOT_USERNAMES
+        .fetch("paid_agent", []).map(&:downcase).to_set
+      non_paid_allowed = allowed.nil? ? nil : (allowed - paid_agent_logins)
       if project&.review_method_enabled?("paid_agent") &&
```
Comment (viamin) [inline app/temporal/activities/scan_paid_prs_activity.rb:667]: This does not make mixed configurations safe yet. `latest` was already chosen from *all* enabled bot logins above, so if a newer clean `paid-agent[bot]` review exists alongside an older Copilot review with unresolved findings, `status` becomes `:clean` and we return `[]` without surfacing the Copilot feedback. In other words, a paid-agent clean review can still clear another bot's outstanding comments. The scanner needs to evaluate paid-agent separately, or at minimum exclude paid-agent reviews from the shared `latest_allowed_bot_review` path when another review bot is enabled.

**Comment by viamin** [comment]
## Agent Update

Commit succeeded. All 5766 tests pass, lint is clean.

Here's a summary of what was done:

**Merge conflicts**: Resolved cleanly — `origin/main` merged into the branch without conflicts.

**Review comment 1 (project.rb:481)** — Fixed: `review_bot_request_login` now delegates to a new `review_bot_request_logins` method that returns **all** enabled bot logins (not just the first). The scanner emits one `review_bot_review_pending` trigger per external bot, and a separate paid-agent trigger. The poll workflow partitions logins into paid-agent (internal agent run) and external bots (GitHub review request API), handling each independently.

**Review comment 2 (scan_paid_prs_activity.rb:598)** — Fixed: Paid-agent logins are now excluded from the shared `latest_allowed_bot_review` path, so a clean paid-agent review can never suppress outstanding findings from another bot like Copilot. Paid-agent clean status is evaluated independently via `paid_agent_clean_review_present?` at the end of `check_review_bot_status`. The `paid_agent_review_clean?` helper was also updated to recognize both clean markers (`<!-- paid-review-clean -->` and `<!-- paid-review-signal: clean -->`).

File: app/temporal/activities/scan_paid_prs_activity.rb [hunk 13]
```diff
@@ -21,17 +21,20 @@ class ScanPaidPrsActivity < BaseActivity
...
     # distinctive phrase rather than the prefix so we are robust to minor
     # wording changes.
     BODY_ONLY_BOT_CLEAN_COMMENT_PATTERN = /didn'?t find any (?:major )?issues/i
 
-    # paid_agent reviews are posted from regular GitHub accounts (not bot
-    # logins registered in PROVIDER_BOT_USERNAMES), so they are invisible
-    # to review_bot? and enabled_review_bot_logins. The agent includes a
-    # machine-readable HTML comment marker when no major findings remain.
+    # paid_agent reviews are posted from regular GitHub accounts, not the
+    # bot logins registered in PROVIDER_BOT_USERNAMES. The agent includes
```
Comment (viamin) [inline app/temporal/activities/scan_paid_prs_activity.rb]: This comment now contradicts the change: this PR registers `paid-agent` and `paid-agent[bot]` in `PROVIDER_BOT_USERNAMES` (lib/provider_support.rb:205-208), so the claim that paid_agent reviews are "not the bot logins registered in PROVIDER_BOT_USERNAMES" is no longer accurate.

```suggestion
    # paid_agent reviews use bot logins registered in PROVIDER_BOT_USERNAMES,
    # but are evaluated separately from the shared bot-review path (see
    # check_review_bot_status). The agent includes
```

**Review by viamin — COMMENTED** [review]
Well-structured PR that cleanly integrates paid-agent into the auto-merge pipeline. The separation of paid-agent evaluation from the shared bot path in `check_review_bot_status` is a sound design choice. One stale comment to fix and a follow-up suggestion.

File: app/temporal/activities/scan_paid_prs_activity.rb [hunk 14]
```diff
@@ -952,7 +967,8 @@ def review_bot?(login)
     def paid_agent_review_clean?(body)
       return false if body.nil?
 
-      body.include?(PAID_REVIEW_CLEAN_MARKER)
+      body.include?(PAID_REVIEW_CLEAN_MARKER) ||
+        REVIEW_BOT_PAID_CLEAN_PATTERN.match?(body)
     end
 
     # TODO(#918): paid_agent posts reviews from regular GitHub accounts — there is
```
Comment (viamin) [inline app/temporal/activities/scan_paid_prs_activity.rb]: Now that `paid-agent` / `paid-agent[bot]` are registered in `PROVIDER_BOT_USERNAMES`, this TODO is actionable — `paid_agent_clean_review_present?` could filter reviews by the paid-agent bot login instead of checking the latest review from any author. Without filtering, a human review posted after a clean paid-agent review will shadow the clean signal and re-trigger a review-goal run.

Not blocking for this PR since the TODO already tracks it, but worth a follow-up now that the bot identity exists.

**Comment by viamin** [comment]
## Agent Update

Done. Here's a summary of what was accomplished:

**Merge conflicts**: None — branch was already up to date with `origin/main`.

**Review comment 1** (scan_paid_prs_activity.rb:36): Fixed the outdated comment. It previously said paid_agent reviews are "not the bot logins registered in PROVIDER_BOT_USERNAMES" — updated to accurately state they use bot logins in `PROVIDER_BOT_USERNAMES` but are evaluated separately from the shared bot-review path.

**Review comment 2** (scan_paid_prs_activity.rb:974): Implemented the suggested improvement. `paid_agent_clean_review_present?` now filters reviews by the paid-agent bot logins (`paid-agent`, `paid-agent[bot]`) from `PROVIDER_BOT_USERNAMES` instead of checking the latest review from any author. This prevents a human review from shadowing or satisfying the clean signal. Also fixed a bug: the review data uses `:user_login`, not `:user`. Updated all 4 test contexts to use `paid-agent[bot]` as the review author instead of `human-reviewer`.

**Validation**: 116 examples, 0 failures in changed specs. Full rubocop clean (716 files, 0 offenses).

**Review by viamin — COMMENTED** [review]
Generated no new comments. The PR is well-structured: bot identity registration in PROVIDER_BOT_USERNAMES collapses identity detection into existing infrastructure, the paid-agent evaluation path is correctly separated from the shared bot-review path to prevent cross-bot clean-status masking, and the new REVIEW_BOT_PAID_CLEAN_PATTERN / start_review_goal_workflow routing are properly wired end-to-end. Tests cover mixed configurations, deduplication edge cases, and both clean markers. No actionable issues found.

**Comment by viamin** [comment]
## Agent Update

Done. Here's a summary of what was accomplished:

**Merge conflicts resolved** in two files:
1. **`app/temporal/activities/queue_agent_run_activity.rb`** — The branch added `goal` parameter extraction while main added `count_toward_draft_review_round` and `expected_draft_review_count`. Kept both sets of input parameters since they're independent features.
2. **`spec/temporal/activities/queue_agent_run_activity_spec.rb`** — The branch had goal parameter tests while main had draft review round tracking tests. Kept both test groups.

**Validation**: 5830 examples, 0 failures, 2 pending. Rubocop clean (719 files, 0 offenses).

**Conversation comments**: All agent update comments were status reports of previously completed work — no outstanding actionable items to address.

**Review by viamin — COMMENTED** [review]
Generated no new comments. The code changes are well-structured and thoroughly documented. Key observations:

- Bot identity registration in `PROVIDER_BOT_USERNAMES` is the right approach — it collapses identity detection into existing infrastructure.
- The paid_agent/external-bot separation in `check_review_bot_status` correctly prevents one bot's clean review from masking another's findings in mixed configurations.
- The incremental issue sync watermark logic handles edge cases (truncation, same-second boundaries) carefully.
- The `OrphanBranchReaperJob` has proper idempotency, grace periods, and TOCTOU mitigation.
- The draft review round tracking with legacy fallback is well-guarded against double-counting.
- Set operations in `check_review_bot_status` are safe since both `enabled_review_bot_logins` and `paid_agent_logins` downcase consistently.

**Comment by viamin** [comment]
Closing per decision in #1011 — incremental, focused PRs are the strategy going forward. This umbrella PR conflicted with already-merged work (#993, #921, #927) and overlapped with independent PRs (#1000). Remaining unique pieces (prompt changes, any code not covered by other PRs) should be extracted into their own focused PRs.
